CDP4Dal/Session.cs

```diff
@@ -326,6 +304,17 @@ public async Task Open()
 
             CDPMessageBus.Current.SendMessage(new SessionEvent(this, SessionStatus.BeginUpdate));
             await this.Assembler.Synchronize(dtoThings);
+
+            this.ActivePerson = this.Assembler.Cache.Select(x => x.Value)
+                .Select(lazy => lazy.Value)
+                .OfType<Person>()
+                .SingleOrDefault(pers => pers.ShortName == this.Credentials.UserName);
+
+            if (this.ActivePerson == null)
+            {
+                throw new IncompleteModelException("The Person object that matches the user specified in the Credentials could not be found");

```

Comment:
I think maybe we should invalidate cache in this case and do not allow any operations with data. If this check is passed then probably we do not need to check active person in other "read" operations, because to perform them this "read" should be already performed.